server/bucket_notifications/views.py

```diff
@@ -0,0 +1,153 @@
+import os
+
+from bson.objectid import ObjectId
+from girder import logger
+from girder.api import access
+from girder.api.describe import Description, autoDescribeRoute
+from girder.api.rest import Resource
+from girder.constants import SortDir
+from girder.exceptions import RestException
+from girder.models.assetstore import Assetstore
+from girder.models.folder import Folder
+from girder.models.user import User
+
+from dive_utils.types import AssetstoreModel, GirderModel
+
+from .constants import AssetstoreRuleMarker
+from .models import (
+    GCSNotificationRecord,
+    GCSPushNotificationPayload,
+    NotificationRouterRule,
+)
+
+
+class BucketNotification(Resource):
+    def __init__(self):
+        super(BucketNotification, self).__init__()
+        self.resourceName = "bucket_notifications"
+
+        self.route("POST", ("routing", ":id"), self.set_notification_routing)
+        self.route("GET", ("gcs",), self.gcs_get_records)
+        self.route("POST", ('gcs',), self.gcs_save_record)
```

Comment:
The idea is that we can add S3 suport to this plugin in the future if we want with new handlers.